fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneScanAllEntriesTest.java

```diff
@@ -211,4 +226,260 @@ private void assertIndexEntryPrimaryKeyTuples(Set<Tuple> primaryKeys, RecordCurs
         assertEquals(primaryKeys,
                 indexEntries.stream().map(IndexEntry::getPrimaryKey).collect(Collectors.toSet()));
     }
+
+    @ParameterizedTest
+    @BooleanSource
+    public void indexScanWithEvenRecNoFilterTest(boolean isGrouped) throws Exception {
+        final long seed = 9876543L;
+        final boolean isSynthetic = false;
+
+        final LuceneIndexTestDataModel dataModel =
+                new LuceneIndexTestDataModel.Builder(seed, this::getStoreBuilderFilterOddRecNo, pathManager)
+                .setIsGrouped(isGrouped)
+                .setIsSynthetic(isSynthetic)
+                .setPrimaryKeySegmentIndexEnabled(true)
+                .setPartitionHighWatermark(10)
+                .build();
+
+        // Save records with a filter that only indexes records with even recNo
+        try (FDBRecordContext context = openContext()) {
+            // Save 10 records - only even recNo values should be indexed
+            // Based on LuceneIndexTestDataModel:267, recNo = 1001 + uniqueCounter
+            // So uniqueCounter values will be 1-10, making recNo values 1002-1011
+            // Even recNos: 1002, 1004, 1006, 1008, 1010 (5 records)
```

Comment:
Having to refer to a specific line number in `LuceneIndexTestDataModel` feels like a code smell to me.
Could a different predicate, or pushing more of this down into `LuceneIndexTestDataModel` make this clearer?
Would it be easier to have a predicate on "other" value, and add a `getOtherValue()` to `RecordsUnderTest`?